tests/post/steps/test_sanity.py

```diff
@@ -69,6 +69,33 @@ def test_statefulset_running(host):
     pass
 
 
+@scenario(
+    "../features/sanity.feature",
+    "Control Plane Ingress Controller when MetalLB is disabled",
+)
+def test_cp_ingress_controller_no_metallb(host):
+    pass
+
+
+@scenario(
+    "../features/sanity.feature",
+    "Control Plane Ingress Controller when MetalLB is enabled",
+)
+def test_cp_ingress_controller_metallb(host):
+    pass
+
+
+# }}}
+# Given {{{
+
+
+@given(parsers.parse("MetalLB is {state}"))
+def is_metalb_enabled(host, state):
+    expected = state == "enabled"
+    if expected != utils.get_pillar(host, "networks:control_plane:metalLB:enabled"):
+        pytest.skip("We skip as we run this test only if MetalLB {}".format(state))
```

Comment:
Can't we have this `given` change the current configuration if it doesn't match what's expected (and restore after the test), instead of skipping?

I guess we could add a command-line flag to disable MetalLB tests if it's really required.

Reply:
Yes we could

Reply:
Oh no sorry I read too quickly, this one cannot change it's in `test_sanity` so to me here we only want to check the running pods according to the configuration (which is what we do here, the check is different between if you have MetalLB or not).

Reply:
Indeed, good point, for these scenarios we shouldn't change things (we could document these tests for open-source users wanting to validate their deployment, akin to [`helm test`](https://helm.sh/docs/topics/chart_tests/)).